### ISSUE TITLE
fix reverse readdir for echoes (bruker)

### ISSUE THREAD

**Issue by aTrotier** [MagneticResonanceImaging/MRIReco.jl#186]
https://github.com/MagneticResonanceImaging/MRIReco.jl/commit/dee7b36f74de86b5220d25e1afc2ed897e2368c5#commitcomment-142358273

**Comment by aTrotier** [comment]
How would you prefer to proceed this :
- I can handle that directly in BrukerFiles, just checking if the parameter with positive readout exists which is, to my knowledge, specific to MGE and implement the reverse like you did (I have to do that in PV360 and PV6)
- Create a dedicated function to reverse the lines according to the echo number `function raw = reverseEchoDir(raw)`

**Comment by tknopp** [comment]
I think the first one is better. The conversion should then be done when converting RawAcqData -> AcqData. I would say also the offset correction in the conversion should be done automatically. Currently there is a parameter "OffsetBruker" (https://github.com/MagneticResonanceImaging/MRIReco.jl/blob/master/MRIBase/src/Datatypes/RawAcqData.jl#L283), which should not be named that way since at that point, there is nothing Bruker specific.

**Comment by aTrotier** [comment]
I don't know if the offset is required for all the dataset from clinical scanner but the OffSetBruker is specific because it should be only applied along phase and partition direction (Bruker directly handle the frequency one).

If we have to apply it automatically it should be handle during the conversion from datasets to raw.

or we don't handle that automatically and just create a specific function like 
https://github.com/MagneticResonanceImaging/MRIReco.jl/blob/f70b2e0c9bf7b1f54f9f91b3d978fba5d2fb5041/MRIBase/src/Datatypes/AcqData.jl#L437

+ that parts
```
  if OffsetBruker
    ROT = [[f.profiles[1].head.read_dir...] [f.profiles[1].head.phase_dir...] [f.profiles[1].head.slice_dir...]]
    offset = inv(ROT) * [f.profiles[1].head.position...]
    offset = offset .*[0 1 1]'; @info "read offset not performed (only phase + slice)";
    acq = correctOffset(acq, offset)
  end
```

**Comment by tknopp** [comment]
Ah, ok, thats a good point. Then, we could probably could simply set the offset when converting from BrukerData to RawData in that direction to zero. In the RawAcq -> Acq conversion one would then only do the correction, if the offset is different from zero in the respective direction. In that way, the  RawAcq -> Acq conversion is generic.

The same line of thoughts holds for the ROT. By setting it correct in BrukerFile (like I did for MGE), one can implement this in a generic fashion in the later conversion.

**Comment by aTrotier** [comment]
One of my colleague told me that the offset is generally handle by Siemens in each sequence.

If it is the case we should change the head.position = (0,0,0) during ismrmrd -> raw.
and for Bruker : head.position = (0,posy,posz) 

But we loose the pure correspondance between the mrd format and the raw format ? 
And maybe the offset is not corrected for GE / Philips but we don't know if the .MRD files was converted by siemens/GE/...

**Comment by aTrotier** [comment]
I should really work on that... It is causing a lot of issue later.

This was introduced in order to reconstruct the MGE. I will implement the reverse flag in the Bruker conversion and every line with this flag will be reverted during the raw -> acq conversion

It will also happen for siemens acquisition converted in MRD if the line has the [flag reverse](https://ismrmrd.readthedocs.io/en/latest/mrd_raw_data.html#mrd-acquisitionflags)

@tknopp Are you ok with that ?

**Comment by aTrotier** [comment]
fixed by #202
